internal/terraform/transform_config.go

```diff
@@ -171,6 +172,21 @@ func (t *ConfigTransformer) transformSingle(g *Graph, config *configs.Config) er
 			importTargets: imports,
 		}
 
+		// If a resource is the target of an import block, the import block
+		// must not reference that resource, except in the "to" attribute.
+		for _, ref := range abstract.ImportReferences() {
```

Comment:
We should avoid validation from within transformers when possible. While these are separate blocks, the logic is technically happening within a single resource "node", so I think this could be handled in validation, and return a more accurate diagnostic to the user.